tools/explorer/lib/ontology_wrapper.py

```diff
@@ -105,58 +103,225 @@ def GetFieldsForTypeName(
+    f-scores. The first f-score is the measure of correctly matched required
+    fields in the canonical type and the second f-score is the measure of
+    correctly matched fields regardless of optionality. Adding the two f-scores
+    creates a preference for matches with a higher quantity of matched required
+    canonical fields, and dividing by 2 keeps the range of the function in
+    [0, 100].
 
     Args:
-      concrete_fields: A set of EntityTypeField objects belonging to the
+      concrete_fields: A set of StandardField objects belonging to the
         concrete entity being matched.
       canonical_fields: A set of EntityTypeField objects belonging to an Entity
         Type defined in DBO.
 
     Returns:
-      The weight of the match as a floating point number.
+      A match's score as an integer.
     """
 
-  def _CreateMatch(self, field_list: List[EntityTypeField],
+    required_canonical_fields = {
+        model.StandardizeField(field) for field in canonical_fields
+        if not field.IsOptional()
+    }
+
+    standard_canonical_fields: Set[StandardField] = set()
+    for field in canonical_fields:
+      standard_canonical_fields.add(model.StandardizeField(field))
+
+    matched_fields = len(
+        concrete_fields.intersection(standard_canonical_fields)
+    )
+    matched_required_fields = len(
+        concrete_fields.intersection(required_canonical_fields)
+    )
+
+    total_required_type_fields = len(required_canonical_fields)
+    total_entity_fields = len(concrete_fields)
+    unmatched_entity_fields = len(
+        concrete_fields.difference(standard_canonical_fields)
+    )
+    unmatched_required_fields = len(
+        required_canonical_fields.difference(concrete_fields)
+    )
+
+    if total_entity_fields <= 0:
+      raise ValueError('Concrete field set cannot be empty.')
+
+    total_precision = matched_fields - unmatched_entity_fields
+    total_precision /= total_entity_fields
+
+    if total_required_type_fields <= 0:
+      match_score = total_precision / 2.0
+    else:
+      required_precision = matched_required_fields - unmatched_required_fields
+      required_precision /= total_required_type_fields
+      match_score = (total_precision + required_precision) / 2.0
+    return int((match_score + 1.0) * 50)
+
+  def _CreateMatch(self, field_list: List[StandardField],
                    entity_type: EntityType) -> Match:
-    """Determines the match between: EntityType and EntityTypeFields.
+    """Creates a Match instance for an EntityType object and a list of
+    StandardField objects.
 
     calls _CalculateMatchWeight() on field_list and the set of fields belonging
-    to entity_type. The weight function outputs a weight signifying the
-    closeness of the match and an instance of the Match class is created with
-    field_list, entity_type, and weight as arguments.
+    to entity_type. The scoring function outputs a float signifying the
```

Comment:
is it a float?

Reply:
Nope, thanks for catching that. Changed to an int with teh range.